### ISSUE TITLE
Schema generator crash after update from 4.2.1 to 4.2.3

### ISSUE THREAD

**Issue by xeladejo** [mikro-orm/mikro-orm#961]
**Describe the bug**
Schema generator crash after update from 4.2.1 to 4.2.3

**Stack trace**
```
TypeError: Reduce of empty array with no initial value
    at Array.reduce (<anonymous>)
    at SchemaBuilder.Target.toQuery (/home/alejandro/Proyectos/tattoo-latest/server/node_modules/knex/lib/interface.js:14:8)
    at SchemaGenerator.dump (/home/alejandro/Proyectos/tattoo-latest/server/node_modules/@mikro-orm/knex/schema/SchemaGenerator.js:485:29)
    at SchemaGenerator.getCreateSchemaSQL (/home/alejandro/Proyectos/tattoo-latest/server/node_modules/@mikro-orm/knex/schema/SchemaGenerator.js:46:25)
    at Function.handleSchemaCommand (/home/alejandro/Proyectos/tattoo-latest/server/node_modules/@mikro-orm/cli/commands/SchemaCommandFactory.js:70:44)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**To Reproduce**
1. Upgrade from 4.2.1 to 4.2.3
2. execute `npx mikro-orm schema:create -d`

**Expected behavior**
That Schema generator continue working after upgrade.

**Versions**

| Dependency | Version |
| - | - |
| node | 14.14.0 |
| typescript | 4.0.3 |
| mikro-orm | 4.2.3 |
| your-driver | mysql |

**Comment by B4nan** [comment]
Well, guess what, it works fine for me. So please provide real reproduction :]

**Comment by xeladejo** [comment]
It is more code that only a reproduction, but to test the schema generator I think it is not important. It's just two entities and include the `dockerfile`.

https://github.com/xeladejo/mikro-orm-crash

**Comment by B4nan** [comment]
Ok, this is caused by the latest knex update. When I update knex to from `0.21.6` to `0.21.7` I can observe the same in the ORM tests. So in fact your issue is not upgraded MikroORM, but upgraded knex.

**Comment by lookfirst** [comment]
@B4nan Just ran into this myself. =(

**Comment by B4nan** [comment]
cc @kibertoad 

it looks like the issue is with trying to get query from empty table builder, this previously worked just fine.

**Comment by lookfirst** [comment]
@B4nan Add some tests for this on your end?

**Comment by B4nan** [comment]
The issue is caused by update of knex on the downstream (or upstream? idk :]). Many ORM tests are failing when I upgrade knex to latest. As I said, I locked the dependencies, so this won't happen anymore, and downsteam will always use the versions that are in the ORM lockfile.

**Comment by B4nan** [comment]
I will add that as this issue seems to affect lots of people (I can imagine all the new users seeing this), I will try to ship new version ASAP (later today). For yarn users its easy to workaround via resolutions, but it looks like npm do not support this (just yet, should be planned tho).

**Comment by lookfirst** [comment]
Right, locking versions will fix this.

FYI, it affects my own test suite (which equates to a 'new' user) since my tests reset everything each run.

So, my thinking was that there is some CI tests somewhere that would fail for your release... but I guess that doesn't happen if knex was updated out of band and you aren't running something like hourly tests and as you said... lockfiles fix this.

**Comment by kibertoad** [comment]
@B4nan I'll take a look tonight, thanks!

**Comment by LiamLB** [comment]
Hi, I see the PR is Open for Knex to resolve the issue fully. Thanks, very excited to start using this tool properly. Until that is merged, is there anything I can do to get things up and running? Is there a version of mikro-orm I can pin that will work in the meanwhile?

**Comment by lookfirst** [comment]
@LiamLB Add this to your package.json:

```
  "resolutions": {
    "@mikro-orm/knex/knex": "0.21.6"
  },
```

Assuming you use yarn.

**Comment by LiamLB** [comment]
I usually don't use yarn but am using it for this feature. Doesn't seem to be possible using npm. I guess a manual edit to package-lock.json might be possible for others using npm?

I needed to add an additional resolution to get it working on my side:

```json
    "resolutions": {
        "@mikro-orm/knex/knex": "0.21.6",
        "knex": "0.21.6"
    }
```

Thanks @lookfirst for the quick solution.

**Comment by kibertoad** [comment]
Released in 0.21.8, could someone confirm if published knex version resolves issue for them?

**Comment by kibertoad** [comment]
@LiamLB You can try also just updating knex version, it should resolve the problem.

**Comment by LiamLB** [comment]
> Released in knex 0.21.8, could someone confirm if published knex version resolves issue for them?

Apologies for delayed response. Setting resolutions to that version of Knex does seem to resolve the issues I was having. I dropped database, created migrations and ran them with no issues.

**Comment by lookfirst** [comment]
Also good here. =) Thanks for the fast response!
